### DISCUSSION TITLE
Is there a way to not include certain properties?

### DISCUSSION THREAD

**Discussion by LuiSeidan** [react-querybuilder/react-querybuilder#440] [Q&A]
Hi,
I am using transformQuery, and I would like to know if there is a way to remove certain properties like: 'not', 'path', 'valueSource', 'condition'.

**Comment by jakeboone02** [comment]
If you want a whitelist/allowlist of properties, you can use `JSON.stringify`'s second argument, like below. (This is exactly what the ["json_without_ids" option for `formatQuantity`](https://react-querybuilder.js.org/docs/api/export#json-without-identifiers) does.)

For a blacklist/denylist, see [this tip page](https://react-querybuilder.js.org/docs/tips/adding-removing-query-properties).

```js
JSON.stringify(
  {
    id: "7f8dda6a-81b4-46cc-b4a7-304d62f29c6a",
    combinator: "and",
    not: false,
    rules: [
      {
        id: "6f90f580-139f-4803-80d1-91f0a7370b03",
        field: "firstName",
        value: "Stev",
        operator: "beginsWith",
      },
      {
        id: "1b32f697-24a3-4cbc-bd2e-caa9dbe814cb",
        field: "lastName",
        value: "Vai, Vaughan",
        operator: "in",
      },
    ],
  },
  ["combinator", "rules", "field", "operator", "value"]
);
```

Output:

```
{"combinator":"and","rules":[{"field":"firstName","operator":"beginsWith","value":"Stev"},{"field":"lastName","operator":"in","value":"Vai, Vaughan"}]}
```

To turn that back into a JSON object, `JSON.parse` the result.
